serviceWorker.js

```diff
@@ -0,0 +1,214 @@
+      return cache.addAll(files)
+      .then(() => {
+        console.info('All files are cached');
+        return self.skipWaiting(); //To forces the waiting service worker to become the active service worker
+      })
+      .catch((error) =>  {
+        console.error('Failed to cache', error);
+      })
+    })
+  );
+});
+
+/*
+  FETCH EVENT: triggered for every request made by index page, after install.
+*/
+
+//Adding `fetch` event listener
+self.addEventListener('fetch', (event) => {
+  console.info('Event: Fetch');
+
+  var request = event.request;
+
+  //Tell the browser to wait for newtwork request and respond with below
+  event.respondWith(
+    //If request is already in cache, return it
+    caches.match(request).then((response) => {
+      if (response) {
+        return response;
+      }
+
+      // // Checking for navigation preload response
+      // if (event.preloadResponse) {
+      //   console.info('Using navigation preload');
+      //   return response;
+      // }
+
+      //if request is not cached or navigation preload response, add it to cache
+      return fetch(request).then((response) => {
+        var responseToCache = response.clone();
+        caches.open(cacheName).then((cache) => {
+            cache.put(request, responseToCache).catch((err) => {
+              console.warn(request.url + ': ' + err.message);
+            });
+          });
+
+        return response;
+      });
+    })
+  );
+});
+
+/*
+  ACTIVATE EVENT: triggered once after registering, also used to clean up caches.
+*/
+
+//Adding `activate` event listener
+self.addEventListener('activate', (event) => {
+  console.info('Event: Activate');
+
+  //Navigation preload is help us make parallel request while service worker is booting up.
+  //Enable - chrome://flags/#enable-service-worker-navigation-preload
+  //Support - Chrome 57 beta (behing the flag)
+  //More info - https://developers.google.com/web/updates/2017/02/navigation-preload#the-problem
+
+  // Check if navigationPreload is supported or not
+  // if (self.registration.navigationPreload) {
+  //   self.registration.navigationPreload.enable();
+  // }
+  // else if (!self.registration.navigationPreload) {
+  //   console.info('Your browser does not support navigation preload.');
+  // }
+
+  //Remove old and unwanted caches
+  event.waitUntil(
+    caches.keys().then((cacheNames) => {
+      return Promise.all(
+        cacheNames.map((cache) => {
+          if (cache !== cacheName) {
+            return caches.delete(cache); //Deleting the old cache (cache v1)
+          }
+        })
+      );
+    })
+    .then(function () {
+      console.info("Old caches are cleared!");
+      // To tell the service worker to activate current one
+      // instead of waiting for the old one to finish.
+      return self.clients.claim();
+    })
+  );
+});
+
+/*
+  PUSH EVENT: triggered everytime, when a push notification is received.
```

Comment:
I don't want push notifications, so this can be removed.

Reply:
Ok :-)